### PR TITLE
Introduce DI into Nethermind Initialization + Plugins

### PR DESCRIPTION
**Pull request by smartprogrammer93**
I know that i could be dreaming to introduce something like this, but i am seeing a lot of people on our team dreaming and their dreams are almost coming true, so i took the initiative after i saw @jmederosalvarado PR here:
https://github.com/NethermindEth/nethermind/pull/5494

let me know what you guys think

P.S : this builds on top of what @jmederosalvarado was doing in the PR.

#### What types of changes does your code introduce?
- [x] Refactoring

### CODE REVIEW

File: src/Nethermind/Nethermind.Api/Factories/IApiComponentFactory.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace Nethermind.Api.Factories;
+
+public interface IApiComponentFactory<out T>
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Api/Factories/IApiComponentFactory.cs:3]: Shouldnt the DI already have mechanism for factory, meaning this type is unnecessary? Like [this](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addsingleton?view=dotnet-plat-ext-7.0#microsoft-extensions-dependencyinjection-servicecollectionserviceextensions-addsingleton(microsoft-extensions-dependencyinjection-iservicecollection-system-type-system-func((system-iserviceprovider-system-object))))

Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Api/Factories/IApiComponentFactory.cs:3]: Yes, i mentioned that in one of my comments

**Comment by smartprogrammer93** [comment]
@LukaszRozmej of course not. using `_api.Services.GetService<ISometing>` is an anti-pattern. i am only using it temporarily till all the services become registered then we will not have to use `GetService` ever again

**Comment by LukaszRozmej** [comment]
> @LukaszRozmej of course not. using `_api.Services.GetService<ISometing>` is an anti-pattern. i am only using it temporarily till all the services become registered then we will not have to use `GetService` ever again

Ok so it is hard to judge without seeing an example of how the goal would look here?

**Comment by asdacap** [comment]
I think you need to do more in this case.

**Comment by smartprogrammer93** [comment]
@LukaszRozmej @asdacap yes. Doing more will surely show its usefulness better

**Comment by smartprogrammer93** [comment]
Closing in favor of #6483

**Pull request closed without merging**